### ISSUE TITLE
Bouncy castle runtime issue 

### ISSUE THREAD

**Issue by sahil994** [babylonhealth/certificate-transparency-android#33]
java.lang.NoClassDefFoundError: Failed resolution of: Lorg/bouncycastle/util/io/pem/PemReader;
        at com.babylon.certificatetransparency.internal.utils.PublicKeyFactory.fromPemString(PublicKeyFactory.kt:47)
        at com.babylon.certificatetransparency.internal.loglist.GooglePublicKeyKt.<clinit>(GooglePublicKey.kt:21)
        at com.babylon.certificatetransparency.internal.loglist.GooglePublicKeyKt.getGoogleLogListPublicKey(GooglePublicKey.kt:21)
        at com.babylon.certificatetransparency.internal.loglist.parser.LogListVerifier.<init>(LogListVerifier.kt:28)
        at com.babylon.certificatetransparency.internal.loglist.parser.RawLogListToLogListResultTransformer.<init>(RawLogListToLogListResultTransformer.kt:33)
        at com.babylon.certificatetransparency.internal.loglist.LogListDataSourceFactory.create(LogListDataSourceFactory.kt:34)
        at com.babylon.certificatetransparency.internal.verifier.CertificateTransparencyBase.<init>(CertificateTransparencyBase.kt:66)
        at com.babylon.certificatetransparency.internal.verifier.CertificateTransparencyHostnameVerifier.<init>(CertificateTransparencyHostnameVerifier.kt:40)
        at com.babylon.certificatetransparency.CTHostnameVerifierBuilder.build(CTHostnameVerifierBuilder.kt:214)
        at com.ikea.network.NetworkCaller.execute(NetworkCaller.java:98)
        at com.ikea.shared.config.service.ConfigService.fetchConfig(ConfigService.java:164)
        at com.ikea.shared.config.service.ConfigService.performFetchConfigOperation(ConfigService.java:202)
        at com.ikea.shared.config.service.ConfigService.access$000(ConfigService.java:43)
        at com.ikea.shared.config.service.ConfigService$2.run(ConfigService.java:179)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "org.bouncycastle.util.io.pem.PemReader" on path: DexPathList[[zip file "/system/framework/com.google.android.maps.jar", zip file "/data/app/com.ikea.ksa.kompis-1/base.apk"],nativeLibraryDirectories=[/data/app/com.ikea.ksa.kompis-1/lib/arm, /data/app/com.ikea.ksa.kompis-1/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.babylon.certificatetransparency.internal.utils.PublicKeyFactory.fromPemString(PublicKeyFactory.kt:47) 
        at com.babylon.certificatetransparency.internal.loglist.GooglePublicKeyKt.<clinit>(GooglePublicKey.kt:21) 
        at com.babylon.certificatetransparency.internal.loglist.GooglePublicKeyKt.getGoogleLogListPublicKey(GooglePublicKey.kt:21) 
        at com.babylon.certificatetransparency.internal.loglist.parser.LogListVerifier.<init>(LogListVerifier.kt:28) 
        at com.babylon.certificatetransparency.internal.loglist.parser.RawLogListToLogListResultTransformer.<init>(RawLogListToLogListResultTransformer.kt:33) 
        at com.babylon.certificatetransparency.internal.loglist.LogListDataSourceFactory.create(LogListDataSourceFactory.kt:34) 
        at com.babylon.certificatetransparency.internal.verifier.CertificateTransparencyBase.<init>(CertificateTransparencyBase.kt:66) 
        at com.babylon.certificatetransparency.internal.verifier.CertificateTransparencyHostnameVerifier.<init>(CertificateTransparencyHostnameVerifier.kt:40) 
        at com.babylon.certificatetransparency.CTHostnameVerifierBuilder.build(CTHostnameVerifierBuilder.kt:214) 
        at com.ikea.network.NetworkCaller.execute(NetworkCaller.java:98) 
        at com.ikea.shared.config.service.ConfigService.fetchConfig(ConfigService.java:164) 
        at com.ikea.shared.config.service.ConfigService.performFetchConfigOperation(ConfigService.java:202) 
        at com.ikea.shared.config.service.ConfigService.access$000(ConfigService.java:43) 
        at com.ikea.shared.config.service.ConfigService$2.run(ConfigService.java:179) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
        at java.lang.Thread.run(Thread.java:761)

**Comment by mattmook** [comment]
Hi @sahil994 would it be possible to see what version of bouncy castle the app is compiling with by executing something like `./gradlew app:dependencies`?

**Comment by lubbo** [comment]
We had the same problem.
We figured out we excluded some bouncycastle libs in the App gradle.

**Comment by mattmook** [comment]
hi @lubbo, could you expand on your solution a little more? Unfortunately I've not been able to reproduce the original bug so I wasn't even quite sure what was causing the issue in the first place, my own initial thought being an issue with Multidex.

**Comment by lubbo** [comment]
Actually it wasn’t a bug for me.
We were using 2 libs (certificate-transparency-android and another one). They imported the same bouncycastle lib as dependency. That generated a duplicate import.
To resolve the problem we excluded bouncycastle lib dependency for c-t-android in the gradle config.
Later we removed the other lib from the project keeping only c-t-android, but we forgot to remove also the exclusion in the gradle config.
So c-t-android dependency to bouncycastle was not found!
Removing the exclusion from gradle config made it working.
